### PR TITLE
Fix Konnectivity with NodePort / LB expose strategy

### LINKED ISSUE #8278: Fix Konnectivity with nodePort / LB expose strategy
Right now, Konenctivity agent in user cluster cannot connect to Konnectivity server when nodeport/LB expose strategy is used.

Make sure that Konenctivity works with any expose strategy.

### PR DESCRIPTION
**Pull request by rastislavs**
**What this PR does / why we need it**:
Konnectivity agent in the user cluster is able to connect to the Konnectivity server in seed only if the Tunneling expose strategy is used. This PR fixes it for the NodePort & LoadBalancer expose strategies as well.

Also updates the `hack/run-user-cluster-controller-manager.sh` so that it works both with and without Konnectivity and with any expose strategy.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #8278 

**Special notes for your reviewer**: `IsServerCertificateValidForAllOf` call in Konnectivity `TLSServingCertificateCreator` had to be moved down, below all certificate alt names are constructed. Otherwise the certificate is re-issued on each reconcile.

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: cmd/user-cluster-controller-manager/main.go [hunk 1]
```diff
@@ -162,9 +166,12 @@ func main() {
 	if err != nil {
 		log.Fatalw("Failed parsing clusterURL", zap.Error(err))
 	}
-	if runOp.openvpnServerPort == 0 {
+	if !runOp.isKonnectivityEnabled && runOp.openvpnServerPort == 0 {
 		log.Fatal("-openvpn-server-port must be set")
 	}
```
Comment (PratikDeoghare) [inline cmd/user-cluster-controller-manager/main.go:171]: Did you mean `runOp.konnectivityServerPort` here?

Comment (rastislavs) [reply cmd/user-cluster-controller-manager/main.go:171]: Nope, `konnectivityServerPort` is defaulted to 6443 if not set.
I just wanted to NOT check for `openvpnServerPort` if konnectivity is enabled.

**Review by PratikDeoghare — APPROVED** [review]

**Review by embik — APPROVED** [review]
/approve
/lgtm

**Pull request merged**